railgun/internal/ctrlutils/ingress-status.go

```diff
@@ -0,0 +1,253 @@
+package ctrlutils
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"sort"
+	"strings"
+
+	"github.com/go-logr/logr"
+	"github.com/kong/deck/file"
+	"github.com/kong/kubernetes-ingress-controller/pkg/sendconfig"
+	"github.com/prometheus/common/log"
+
+	"sync"
+
+	apiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	knative "knative.dev/networking/pkg/apis/networking/v1alpha1"
+	knativeversioned "knative.dev/networking/pkg/client/clientset/versioned"
+	knativeApis "knative.dev/pkg/apis"
+	"knative.dev/pkg/network"
+)
```

Comment:
nit: for consistency we should use the `goimports -local` rule of separating std, 3rd party and local into whitespace separated import groups:

```suggestion
import (
	"context"
	"fmt"
	"net"
	"sort"
	"strings"
	"sync"

	"github.com/go-logr/logr"
	"github.com/kong/deck/file"
	"github.com/prometheus/common/log"
	apiv1 "k8s.io/api/core/v1"
	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
	clientset "k8s.io/client-go/kubernetes"
	"k8s.io/client-go/rest"
	knative "knative.dev/networking/pkg/apis/networking/v1alpha1"
	knativeversioned "knative.dev/networking/pkg/client/clientset/versioned"
	knativeApis "knative.dev/pkg/apis"
	"knative.dev/pkg/network"

	"github.com/kong/kubernetes-ingress-controller/pkg/sendconfig"
)
```